r/R/dataset.R

```diff
@@ -15,6 +15,30 @@
 # specific language governing permissions and limitations
 # under the License.
 
+#' @usage NULL
+#' @format NULL
+#' @rdname Source
+#' @export
+DatasetFactory <- R6Class("DatasetFactory", inherit = Object,
```

Comment:
Should `DatasetFactory` be the only way to create a dataset? I.e. should `Dataset$create()` be removed, or maybe it should call `DatasetFactory$create()`?

Reply:
The user can still use `SourceFactory$Finish` to get `Source`, so I'd say we keep `Dataset$create` but recommend in the documentation to use `DatasetFactory`.

Reply:
Still sounds like dead code to me. Why would I call `SourceFactory$Finish` myself?